tests/oper.rs

```diff
@@ -271,6 +271,45 @@ fn fold_and_sum() {
     }
 }
 
+#[test]
+fn fold_and_prod() {
+    let a = Array::linspace(0., 127., 128).into_shape((8, 16)).unwrap();
+    assert_approx_eq(a.fold(1., |acc, &x| acc * x), a.scalar_prod(), 1e-5);
+
+    // test different strides
+    let max = 8 as Ixs;
+    for i in 1..max {
+        for j in 1..max {
+            let a1 = a.slice(s![..;i, ..;j]);
+            let mut prod = 1.;
+            for elt in a1.iter() {
+                prod *= *elt;
+            }
+            assert_approx_eq(a1.fold(1., |acc, &x| acc * x), prod, 1e-5);
+            assert_approx_eq(prod, a1.scalar_prod(), 1e-5);
+        }
+    }
+
+    // skip a few elements
+    let max = 8 as Ixs;
+    for i in 1..max {
```

Comment:
This block of the test isn't necessary. (It's just testing the `fold` method, not `scalar_prod`, and `fold` is already tested in `fold_and_sum`.)

Reply:
Aha! I was already wondering what the point of that test was in `fold_and_sum`. Deleted